x/rollapp/keeper/block_height_to_finalization_queue.go

```diff
@@ -79,15 +74,39 @@ func (k Keeper) GetBlockHeightToFinalizationQueue(
 // RemoveBlockHeightToFinalizationQueue removes a blockHeightToFinalizationQueue from the store
 func (k Keeper) RemoveBlockHeightToFinalizationQueue(
 	ctx sdk.Context,
-	finalizationHeight uint64,
+	creationHeight uint64,
 
 ) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlockHeightToFinalizationQueueKeyPrefix))
 	store.Delete(types.BlockHeightToFinalizationQueueKey(
-		finalizationHeight,
+		creationHeight,
 	))
 }
 
+// GetPendingFinalizationQueue returns the blockHeightToFinalizationQueue starting from the input height till the last queue with batches not yet finalized
+func (k Keeper) GetPendingFinalizationQueue(ctx sdk.Context, height uint64) (list []types.BlockHeightToFinalizationQueue) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.BlockHeightToFinalizationQueueKeyPrefix))
+	heightKey := types.BlockHeightToFinalizationQueueKey(height + 1)
+	iterator := sdk.KVStoreReversePrefixIterator(store, heightKey)
+	defer iterator.Close() // nolint: errcheck
+
+	for ; iterator.Valid(); iterator.Next() {
+		var val types.BlockHeightToFinalizationQueue
+		k.cdc.MustUnmarshal(iterator.Value(), &val)
+		stateInfoIndex := val.FinalizationQueue
+		stateInfo, found := k.GetStateInfo(ctx, stateInfoIndex[0].RollappId, stateInfoIndex[0].Index)
+		if !found {
+			continue
+		}
+		if stateInfo.Status == types.STATE_STATUS_FINALIZED {
```

Comment:
how a fiinalized stateInfo can be found in the pending queue?

Reply:
I understand that, but how finalized state can be in the `ToFinalizationQueue`?
doesn't the `heightKey` in the iterator sets the limit? why u need to check stuff on the state object when getting from store?

Reply:
yes, exactly. i dont use the key as limit, because the disputeperiod can change, and the finalization time can change in the future. i check the state object to see if the state infos in that queue are finalized (i assume all state infos are in the same state)

Reply:
this check has been removed because there is no need for the check because queues are removed after finalization